src/screens/hardware-wallets/components/ErrorSheet.tsx

```diff
@@ -1,50 +1,75 @@
-// import { CancelButton } from '@/screens/hardware-wallets/components/CancelButton';
 import { Layout } from '@/screens/hardware-wallets/components/Layout';
-// eslint-disable-next-line no-restricted-imports
-import { RouteProp, useRoute } from '@react-navigation/core';
-import { useDimensions } from '@/hooks';
+import { useDimensions, useLedgerConnect } from '@/hooks';
 import { Source } from 'react-native-fast-image';
 import ledgerNanoUnlock from '@/assets/ledger-nano-unlock.png';
 import ledgerNanoEthApp from '@/assets/ledger-nano-eth-app.png';
 import { ImgixImage } from '@/components/images';
 import { TRANSLATIONS } from '@/screens/hardware-wallets/constants';
 import { LEDGER_ERROR_CODES } from '@/utils/ledger';
-import { useLedgerConnect } from '@/hooks/useLedgerConnect';
 import { useNavigation } from '@/navigation';
+import Routes from '@/navigation/routesNames';
+import { Alert } from 'react-native';
 
 const IMAGE_ASPECT_RATIO = 1.547;
 const IMAGE_LEFT_OFFSET = 36;
 
-export type PairHardwareWalletErrorSheetParams = {
-  errorType: LEDGER_ERROR_CODES.OFF_OR_LOCKED | LEDGER_ERROR_CODES.NO_ETH_APP;
+type ErrorSheetProps = {
   deviceId?: string;
 };
 
-type RouteParams = {
-  PairHardwareWalletErrorSheetParams: PairHardwareWalletErrorSheetParams;
+export type ErrorSheetRouteParams = {
+  ErrorSheetProps: ErrorSheetProps;
 };
 
-export const PairHardwareWalletErrorSheet = () => {
-  const route = useRoute<
-    RouteProp<RouteParams, 'PairHardwareWalletErrorSheetParams'>
-  >();
+export const ErrorSheet = ({
+  deviceId,
+  type,
+}: {
+  deviceId?: string;
+  type: LEDGER_ERROR_CODES.OFF_OR_LOCKED | LEDGER_ERROR_CODES.NO_ETH_APP;
+}) => {
+  const { dangerouslyGetParent, navigate } = useNavigation();
   const { width: deviceWidth } = useDimensions();
-  const { goBack } = useNavigation();
 
   const imageWidth = deviceWidth - IMAGE_LEFT_OFFSET;
   const imageHeight = imageWidth / IMAGE_ASPECT_RATIO;
 
-  const errorType = route?.params?.errorType;
+  const successCallback = useCallback(() => {
+    dangerouslyGetParent()?.goBack(); // to blind signing sheet
+    // or nav to tx details (?) if confirming a tx
+  }, [dangerouslyGetParent]);
 
-  useLedgerConnect({
-    readyForPolling: !!route?.params?.deviceId,
-    deviceId: route?.params?.deviceId || '',
-    successCallback: () => {
-      goBack();
+  const errorCallback = useCallback(
+    (errorType: LEDGER_ERROR_CODES) => {
+      console.log('error callback', errorType);
+      if (errorType === LEDGER_ERROR_CODES.OFF_OR_LOCKED) {
+        if (type === LEDGER_ERROR_CODES.NO_ETH_APP) {
+          navigate(Routes.PAIR_HARDWARE_WALLET_LOCKED_ERROR_SHEET, {
+            params: deviceId,
+          });
+        }
+      } else if (errorType === LEDGER_ERROR_CODES.NO_ETH_APP) {
+        if (type === LEDGER_ERROR_CODES.OFF_OR_LOCKED) {
+          navigate(Routes.PAIR_HARDWARE_WALLET_ETH_APP_ERROR_SHEET, {
+            params: deviceId,
+          });
+        }
+      } else {
```

Comment:
Can get locked in these if statements and nothing will happen. That desirable? 

Seems like you pry want only one level of conditional logic and do two clauses, so you can fall back to the `else` `Alert`, like:

```javascript
} else if (errorType === LEDGER_ERROR_CODES.NO_ETH_APP && type === LEDGER_ERROR_CODES.OFF_OR_LOCKED) {
```